server/app/utils.py

```diff
@@ -405,3 +412,47 @@ def check_user(user_key):
         user_key = user_key.id()
 
     deferred.defer(deferred_check_user, user_key)
+
+######################
+# Autograder actions #
+######################
+
+def add_taskqueue(submission):
+    q = taskqueue.Queue("pull-queue")
+    backup = submission.backup.get()
+    submission_contents = backup.get_messages().get("file_contents")
+    sub_id = str(submission.key.id())
+
+    tasks = []
+    tasks.append(taskqueue.Task( payload = str(submission_contents), method = "PULL", tag = sub_id))
+    q.add(tasks)
+
+def add_all_taskqueue(course, assign_key):
+    q = taskqueue.Queue("pull-queue")
+    students = course.students()  
+    final_submissions = [student.get().get_final_submission(assign_key) for student in students]
+    submissions = [final.submission.get() for final in final_submissions]
+    backups = [(sub.backup.get(), str(sub.key.id())) for sub in submissions]
+
+    tasks = []
+    for backup, sub_id in backups:
+        submission_contents = backup.get_messages().get("file_contents")
+        tasks.append(taskqueue.Task( payload = submission_contents, method = "PULL", tag = sub_id))
+    q.add(tasks)
+
+def lease_tasks(): 
```

Comment:
The function itself leases all of the files and returns them to the caller.

I'm not sure what exactly this is used for in the architecture. It is used [here](https://github.com/Cal-CS-61A-Staff/ok/pull/438/files#diff-bb50eb2e4a13c0bf26e75a3a8442adb2R312), but that calls the `send` function, which isn't actually defined anywhere. That import is commented out, and the file that it imports the function from (`http_requests.py`) is commented out as well. Maybe this was some test code they used that isn't used in the actual functionality? I'm not sure. 

This doesn't seem to be in the original "flow" of how the autograder seems to work. Something isn't supposed to lease all the tasks, at least in my understanding...